node/src/database/db_migrations/db_migrator.rs

```diff
@@ -0,0 +1,808 @@
+    fn list_validation_check_works_for_badly_ordered_migrations_when_inside() {
+        let fake_one = DBMigrationRecordMock::default().old_version_result(6);
+        let fake_two = DBMigrationRecordMock::default().old_version_result(2);
+        let list: &[&dyn DatabaseMigration] = &[&Migrate_0_to_1, &fake_one, &fake_two];
+
+        let _ = list_validation_check(list);
+    }
+
+    #[test]
+    #[should_panic(expected = "The list of database migrations is not ordered properly")]
+    fn list_validation_check_works_for_badly_ordered_migrations_when_at_the_end() {
+        let fake_one = DBMigrationRecordMock::default().old_version_result(1);
+        let fake_two = DBMigrationRecordMock::default().old_version_result(3);
+        let list: &[&dyn DatabaseMigration] = &[&Migrate_0_to_1, &fake_one, &fake_two];
+
+        let _ = list_validation_check(list);
+    }
+
+    fn list_validation_check<'a>(list_of_migrations: &'a [&'a (dyn DatabaseMigration + 'a)]) {
+        let begins_at_version = list_of_migrations[0].old_version();
+        let iterator = list_of_migrations.iter();
+        let ending_sentinel = &DBMigrationRecordMock::default()
+            .old_version_result(begins_at_version + iterator.len())
+            as &dyn DatabaseMigration;
+        let iterator_shifted = list_of_migrations
+            .iter()
+            .skip(1)
+            .chain(once(&ending_sentinel));
+        iterator.zip(iterator_shifted).for_each(|(first, second)| {
+            assert!(
+                two_numbers_are_sequential(first.old_version(), second.old_version()),
+                "The list of database migrations is not ordered properly"
+            )
+        });
+    }
+
+    fn two_numbers_are_sequential(first: usize, second: usize) -> bool {
+        (first + 1) == second
+    }
+
+    #[test]
+    fn list_of_migrations_is_correctly_ordered() {
+        let _ = list_validation_check(DbMigratorReal::list_of_migrations());
+        //success if no panicking
+    }
+
+    #[test]
+    fn list_of_migrations_ends_on_the_current_version() {
+        let last_entry = DbMigratorReal::list_of_migrations().into_iter().last();
+
+        let result = last_entry.unwrap().old_version();
+
+        assert!(two_numbers_are_sequential(result, CURRENT_SCHEMA_VERSION))
+    }
+
+    #[test]
+    fn migrate_semi_automated_returns_an_error_from_update_schema_version() {
+        let update_schema_version_params_arc = Arc::new(Mutex::new(vec![]));
+        let mut migration_record = DBMigrationRecordMock::default()
+            .old_version_result(4)
+            .migrate_result(Ok(()));
+        let migration_utilities = DBMigrationUtilitiesMock::default()
+            .make_mig_declaration_utils_result(Box::new(
+                DBMigrateDeclarationUtilitiesMock::default(),
+            ))
+            .update_schema_version_result(Err(Error::InvalidQuery))
+            .update_schema_version_params(&update_schema_version_params_arc);
```

Comment:
Conventionally, the order of the `update_schema_version_X` lines should be reversed